arkouda/numeric.py

```diff
@@ -1722,6 +1725,89 @@ def where(
     return create_pdarray(type_cast(str, repMsg))
 
 
+def matching_shapes(pda_a: pdarray, pda_b: pdarray):
+    """
+    For now, this will only check that two pdarrays have matching sizes.
+    It's included so that we can put placeholders in for the check as
+    we implement n-dimensional array computation.
+
+    Parameters:
+        pda_a : pdarray
+        pda_b : pdarray
+
+    Returns:
+        True if pda_a.size == pda_b.size, else False
+
+    Examples
+    --------
+    >>> a = np.random.randint(0,10,10).astype(np.float64)
+    >>> b = np.random.randint(0,3,10)
+    >>> c = ak.array(a)
+    >>> d = ak.array(b)
+    >>> ak.matching_shapes(c,d)
+    True
+    >>> b = np.random.randint(0,10,5000)
+    >>> d = ak.array(b)
+    >>> ak.matching_shapes(c,d)
+    False
+
+    """
+
+    return pda_a.size == pda_b.size
+
+
+def array_equal(pda_a: pdarray, pda_b: pdarray, equal_nan: bool = False):
+    """
+    Compares two pdarrays for equality.
+    If neither array has any nan elements, then if all elements are pairwise equal,
+    it returns True.
+    If equal_Nan is False, then any nan element in either array gives a False return.
+    If equal_Nan is True, then pairwise-corresponding nans are considered equal.
+
+    Parameters:
+        pda_a : pdarray
+        pda_b : pdarray
+        equal_nan : boolean to determine how to handle nans, default False
+
+    Returns:
+        With string data: False if one array is of type npstr and the other isn't,
+                            True is both are npstr and they match.
```

Comment:
So, I can tell the Return section is not quite right because when I `make doc` on this and then navigate in the web browser, the `Return Type` section is missing.  Also, `and all nans in pda_a correspond to nans in pda_b` is highlighted in a pale blue.  I'm sorry, I know that sphinx is really fussy and it can be frustrating to deal with.

Reply:
This is still not rendering correctly for me.  Perhaps you were going for something like this:
```
    Returns
    -------
    boolean

        With string data:
            False if one array is of type npstr and the other isn't,

            True is both are npstr and they match.

        With numeric data:
            True if neither array has any nan elements, and all elements pairwise equal.

            False if equal_Nan is False, and either array has any nan element.

            True if equal_Nan is True, all non-nan elements are pairwise equal,
            and all nans in pda_a correspond to nans in pda_b

```